src/main/java/org/commcare/formplayer/session/MenuSession.java

```diff
@@ -75,9 +77,6 @@
     public MenuSession(SerializableMenuSession session, InstallService installService,
                        RestoreFactory restoreFactory, String host) throws Exception {
         this.session = session;
-        session.setInstallReference(
-                resolveInstallReference(session.getInstallReference(), session.getAppId(), host, session.getDomain())
-        );
```

Comment:
I removed this since it seems unnecessary to resolve the reference of an existing session. The session is created with an already resolved reference and unless the app ID or domain have changed (which they don't) it will resolve to the same thing anyway.

Reply:
I think menu session serialization is rare anyway. It's only really used for end of form nav, so it seems right that even if there was an update to the app build ID that it would be wrong to follow it, we wouldn't want to update the app in between finishing a form and navigating to the next session state.